src/arproc.py

```diff
@@ -1202,28 +1211,34 @@ def trim(slf, file, det):
     return file[w]
 
 
-
-
-
-def variance_frame(slf, det, sciframe, idx, fitsdict, skyframe=None):
+def variance_frame(slf, det, sciframe, idx, fitsdict, skyframe=None, objframe=None):
     """ Calculate the variance image including detector noise
     Parameters
     ----------
     fitsdict : dict
       Contains relevant information from fits header files
+    objframe : ndarray, optional
+      Model of object counts
     Returns
     -------
     variance image : ndarray
     """
-    scicopy = sciframe.copy()
+    # The effective read noise (variance image)
+    rnoise = rn_frame(slf,det)
     if skyframe is not None:
+        '''
         msgs.warn("arproc.variance_frame: JXP worries the next line could be biased.")
         wfill = np.where(np.abs(skyframe) > np.abs(scicopy))
         scicopy[wfill] = np.abs(skyframe[wfill])
-    # Dark Current noise
-    dnoise = (slf._spect['det'][det-1]['darkcurr'] *
-              float(fitsdict["exptime"][idx])/3600.0)
-    # The effective read noise
-    rnoise = rn_frame(slf,det)
-    # Return
-    return np.abs(scicopy) + rnoise + dnoise
+        '''
+        if objframe is None:
+            objframe = np.zeros_like(skyframe)
```

Comment:
Understood -- thanks for the clarification!

Agreed, let's stick with raw counts for the
boxcar.